memcached.c

```diff
@@ -8422,6 +8457,8 @@ int main (int argc, char **argv) {
 
     /* init settings */
     settings_init();
+    assert(hash_type == MURMUR3_HASH);// if this assertion fails, change the default hash_algorithm
```

Comment:
why is this here when the other asserts are up in the help text?

related; do any tests actually call -h explicitly? A bunch of the supports_* stuff does, but perhaps this could use a singular test for "do the help asserts fire"?

Reply:
`hash_type` is local to the main method, it's not a part of the settings unless we refactor it.
+1 on a test, I will add one.